mmv1/third_party/terraform/services/backupdr/data_source_backup_dr_management_server.go.erb

```diff
@@ -0,0 +1,104 @@
+<% autogen_exception -%>
+package backupdr
+<% unless version == 'ga' -%>
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+	"strings"
+)
+
+func DataSourceGoogleCloudBackupDRService() *schema.Resource {
+
+	dsSchema := tpgresource.DatasourceSchemaFromResourceSchema(ResourceBackupDRManagementServer().Schema)
+	tpgresource.AddRequiredFieldsToSchema(dsSchema, "location")
+
+	return &schema.Resource{
+		Read:   dataSourceGoogleCloudBackupDRServiceRead,
+		Schema: dsSchema,
+	}
+}
+func flattenBackupDRManagementServerName(v interface{}, d *schema.ResourceData, config *transport_tpg.Config) interface{} {
+	return v
+}
+
+func flattenBackupDRManagementServerResourceResp(v interface{}, d *schema.ResourceData, config *transport_tpg.Config) map[string]interface{} {
+	if v == nil {
+		fmt.Printf("Interface is nil: %s", v)
+	}
+	fmt.Printf("Interface is : %s", v)
+	l := v.([]interface{})
+	for _, raw := range l {
+		return flattenBackupDRManagementServerResource(raw, d, config)
+	}
+	return nil
+}
+func flattenBackupDRManagementServerResource(v interface{}, d *schema.ResourceData, config *transport_tpg.Config) map[string]interface{} {
+	original := v.(map[string]interface{})
+	if len(original) == 0 {
+		return nil
+	}
+	transformed := make(map[string]interface{})
+	transformed["type"] = flattenBackupDRManagementServerType(original["type"], d, config)
+	transformed["networks"] = flattenBackupDRManagementServerNetworks(original["networks"], d, config)
+	transformed["oauth2ClientId"] = flattenBackupDRManagementServerOauth2ClientId(original["oauth2ClientId"], d, config)
+	transformed["managementUri"] = flattenBackupDRManagementServerManagementUri(original["managementUri"], d, config)
+	transformed["name"] = flattenBackupDRManagementServerName(original["name"], d, config)
+	return transformed
+}
+
+func dataSourceGoogleCloudBackupDRServiceRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*transport_tpg.Config)
+	userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+	if err != nil {
+		return err
+	}
+	project, err := tpgresource.GetProject(d, config)
+	if err != nil {
+		return err
+	}
+	billingProject := project
+	url, err := tpgresource.ReplaceVars(d, config, "{{BackupDRBasePath}}projects/{{project}}/locations/{{location}}/managementServers")
+	if err != nil {
+		return err
+	}
+	if bp, err := tpgresource.GetBillingProject(d, config); err == nil {
+		billingProject = bp
+	}
+	res, err := transport_tpg.SendRequest(transport_tpg.SendRequestOptions{
+		Config:    config,
+		Method:    "GET",
+		Project:   billingProject,
+		RawURL:    url,
+		UserAgent: userAgent,
+	})
+	if err != nil {
+		return fmt.Errorf("Error constructing id: %s", err)
```

Comment:
Do we think if this error message is bit confusing? Can we just simply use `Error reading ManagementServer:` as the rest?